pkg/registry/core/service/storage/rest_test.go

```diff
@@ -2190,10 +2198,16 @@ func TestServiceRegistryExternalTrafficHealthCheckNodePortAllocation(t *testing.
 // Validate using the user specified nodePort when ExternalTrafficPolicy is set to Local
 // and type is LoadBalancer.
 func TestServiceRegistryExternalTrafficHealthCheckNodePortUserAllocation(t *testing.T) {
-	randomNodePort := generateRandomNodePort()
 	ctx := genericapirequest.NewDefaultContext()
 	storage, registry, server := NewTestREST(t, nil, singleStackIPv4)
 	defer server.Terminate(t)
+
+	svcNodePort := generateRandomNodePort(storage.serviceNodePorts)
+	healthCheckPort := generateRandomNodePort(storage.serviceNodePorts)
+	for healthCheckPort == svcNodePort {
+		healthCheckPort = generateRandomNodePort(storage.serviceNodePorts)
+	}
```

Comment:
why is it important for these ports to be random?

Reply:
because we don't know which port is available.

Reply:
oh do you mean we could simply specify static node port? It makes sense to me.

Reply:
yeah, this is a unit test, I'd expect all ports to be available

Reply:
actually... does this flake mean that the API server is allocating a conflicting port? if so, isn't that a bug we should fix?

Reply:
yeah, the apiserver doesn't take health port into account when allocating service node port. I would try to fix it.